roles/smartgateway/tasks/metrics.yml

```diff
@@ -0,0 +1,107 @@
+---
+# tasks file for smartgateway
+- name: Write out configmap for smart gateway
+  template:
+    dest: /tmp/configmap.yaml
+    src: "{{ service_type }}-configmap.yaml.j2"
+
+- name: Smart gateway ConfigMap
+  k8s:
+    src: /tmp/configmap.yaml
+
+- name: Start the smart gateway
+  k8s:
+    resource_definition:
+      apiVersion: apps.openshift.io/v1
+      kind: DeploymentConfig
+      metadata:
+        name: '{{ meta.name }}-smartgateway'
+        namespace: '{{ meta.namespace }}'
+      spec:
+        replicas: '{{ size }}'
+        template:
+          metadata:
+            labels:
+              app: prometheus-{{ meta.name }}
+          spec:
+            securityContext:
+              nonroot: true
+            containers:
+            - name: smart-gateway
+              image: '{{ container_image_path }}'
+              volumeMounts:
+              - name: sg-config
+                mountPath: /config
+              args:
+                - '-config=/config/smartgateway_config.json'
+                - '-uname=$(MY_POD_NAME)'
+                - '-servicetype={{ service_type }}'
+              env:
+                - name: MY_POD_NAME
+                  valueFrom:
+                    fieldRef:
+                      apiVersion: v1
+                      fieldPath: metadata.name
+              ports:
+              - name: smart-gateway
+                containerPort: 8081
+            volumes:
+            - name: sg-config
+              configMap:
+                name: "{{ meta.name }}-smartgateway"
+            nodeSelector:
+              node: "{{ meta.name }}"
+              application: sa-telemetry
+
+- name: Service for the smart gateway
+  k8s:
+    resource_definition:
+      apiVersion: v1
+      kind: Service
+      metadata:
+        name: '{{ meta.name }}-smartgateway'
+        namespace: '{{ meta.namespace }}'
+        labels:
+          smartgateway: '{{ meta.name }}'
+      spec:
+        ports:
+        - name: smart-gateway
+          port: 8081
+          targetPort: 8081
+          protocol: TCP
+        selector:
+          app: prometheus-{{ meta.name }}
+
+- name: Service monitor for smart gateway
```

Comment:
This file is not very DRY. If you were to move everything below this line to it's own file (one file per resource is a pretty common pattern) then this file would differ from events.yml by about six lines, so roughly >90% overlap

Reply:
> Not worth blocking on since nothing looks dangerous or broken, but I do wonder if that events service will work properly without a selector. If I get a chance to test this myself I might approve if I find it works.

I'm going to investigate and see if I can clean this up. I knew some of this was coming, but wanted to get the PR pushed in a known working state. I'm going to iterate on it some more and see if we can get it closer to our idealized state. Thanks!